### ISSUE TITLE
petri.com, SingleFile freezes and provokes high CPU usage

### ISSUE THREAD

**Issue by abolibibelot1980** [gildas-lormeau/SingleFile#428]
**Describe the bug**
When saving this page with SingleFile :
https://petri.com/turn-windows-10-ntfs-case-sensitivity
the operation froze and the CPU usage of one of Firefox's processes increased abnormally (to 13% on a 4 cores / 8 threads CPU which means that a virtual core was saturated), after a while a message appeared on the top of the page saying that a page was slowing down the browser, and the operation seemed to stall until I clicked on the Stop button (“L'arrêter” in french on the screenshot), then the normal process in 3 steps started and the page could be saved, although it might be incomplete. Since near the bottom of the page there was a “loading comments” message with a circle below which wouldn't stop spinning I tried disabling all other extensions which may prevent the proper loading of some elements of that page : NoScript, uBlock, AdBlock -- to no avail.
Save Page WE, another extension with the same purpose which I now use in those rare situations where SingleFile doesn't work properly (as it's less customizable), worked fine and saved a 895KB file. SingleFile saved a 370KB file on the first attempt and 375KB on the second attempt (same behaviour on both attempts, after reloading the page repeatedly and disabling extensions as mentioned above).

**Screenshots**
![petri com SingleFile slow down browser](https://user-images.githubusercontent.com/44624604/83588520-df915880-a551-11ea-8b34-f48b5768f2c4.png)

**Environment**
 - Windows 7 Pro SP1
 - Firefox 76.0.1 64 bits

**Comment by gildas-lormeau** [comment]
Thank you for having reported the issue. This is related to the option "Save deferred images". Disabling it fixes the issue. I'll try to see what's wrong.

**Comment by gildas-lormeau** [comment]
Sorry, I was totally wrong. This issue was related to the UI elements of SingleFile, i.e. the shadow displayed on the top of the page and the logs at the bottom left. I needed to optimize the code a little bit to fix it. I don't know why on this page the code was much slower than on other pages though. 
The fix is available in the version 1.17.46 I just uploaded. This version will be available for download in the coming days, after having been reviewed by Mozilla
